app/controllers/application_controller.rb

```diff
@@ -96,26 +94,26 @@ def store_user_location!
     def set_locale
       I18n.locale = current_user.try(:locale) || locale_param ||
                     cookie_locale_param || I18n.default_locale
-      unless user_signed_in?
-        cookies[:locale] = I18n.locale
-      end
+      return if user_signed_in?
+
+      cookies[:locale] = I18n.locale
     end
 
     def store_interaction
-      return if controller_name.in?(['sessions', 'administration', 'users',
-                                     'events', 'interactions', 'profile',
-                                     'clickers', 'clicker_votes', 'registrations'])
-      return if controller_name == 'main' && action_name == 'home'
-      return if controller_name == 'tags' && action_name.in?(['fill_tag_select',
-                                                              'fill_course_tags'])
+      return if controller_name.in?(["sessions", "administration", "users",
+                                     "events", "interactions", "profile",
+                                     "clickers", "clicker_votes", "registrations"])
+      return if controller_name == "main" && action_name == "home"
+      return if controller_name == "tags" && action_name.in?(["fill_tag_select",
+                                                              "fill_course_tags"])
 
       study_participant = current_user.anonymized_id if current_user.study_participant
       # as of Rack 2.0.8, the session_id is wrapped in a class of its own
       # it is not a string anymore
       # see https://github.com/rack/rack/issues/1433
       InteractionSaver.perform_async(request.session_options[:id].public_id,
                                      request.original_fullpath,
-                                     request.referrer,
+                                     request.referer,
                                      study_participant)
```

Comment:
Fun fact: The corect spelling is actually referrer, but is not used anymore; [https://en.wikipedia.org/wiki/HTTP_referer#Etymology](https://en.wikipedia.org/wiki/HTTP_referer#Etymology)